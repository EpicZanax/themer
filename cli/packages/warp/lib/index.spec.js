const { render, renderInstructions } = require('./index');
const { colors } = require('../../colors-default');

describe('render', () => {
  const promisedFiles = Promise.all(
    render(colors, {
      _: [],
      c: './colors-default',
      colors: './colors-default',
      t: './warp',
      template: ['./warp'],
      o: '/tmp/themer',
      out: '/tmp/themer',
    }),
  );

  it('should render the expected number of files', async () => {
    const files = await promisedFiles;
    expect(files.length).toBe(2);
  });

  it('should render valid Warp theme yaml', async () => {
    const files = await promisedFiles;
    files.forEach((file) => {
      const contents = file.contents.toString('utf8');
      expect(
        /([ ]+)?((\\w+|[^\\w\\s\\r\\n])([ ]*))?(?:\\r)?(\\n)?/.test(contents),
      ).toBe(true);
    });
  });
});

describe('renderInstructions', () => {
  it('should provide installation instructions', async () => {
    const files = await Promise.all(
      render(colors, {
        _: [],
        c: './colors-default',
        colors: './colors-default',
        t: './warp',
        template: ['./warp'],
        o: '/tmp/themer',
        out: '/tmp/themer',
      }),
    );
    const instructions = renderInstructions(files.map(({ name }) => name));
    expect(instructions).toMatchSnapshot();
  });
});
