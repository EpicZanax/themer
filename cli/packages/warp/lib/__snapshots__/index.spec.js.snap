// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderInstructions should provide installation instructions 1`] = `
"
1. Copy your files (\`themer_colors-default_dark.yaml\` and \`themer_colors-default_light.yaml\`) to \`~/.warp/themes/\`
2. Launch Warp
3. Press \`command\`-\`P\` to open the Command Palette
4. Search and select \`Open Theme Picker\`
5. Search for \`Themer Colors Default Dark\` or \`Themer Colors Default Light\` and select it
6. Enjoy your new theme! âœ¨
"
`;